src/track/beats.h

```diff
@@ -54,124 +62,160 @@ class Beats : public QObject {
+
+    /// Starting from frame, fill the frame numbers of the previous beat
+    /// and next beat.  Either can be -1 if none exists.  If frame refers
+    /// to the location of the beat, the first value is frame, and the second
+    /// value is the next beat position.  Non- -1 values are guaranteed to be
+    /// even.  Returns false if *at least one* frame is -1.  (Can return false
+    /// with one beat successfully filled)
+    bool findPrevNextBeats(FramePos frame,
+            FramePos* pPrevBeatFrame,
+            FramePos* pNextBeatFrame) const;
+
+    /// Starting from frame, return the frame number of the closest beat
+    /// in the track, or -1 if none exists.  Non- -1 values are guaranteed to be
+    /// even.
+    FramePos findClosestBeat(FramePos frame) const;
+
+    /// Find the Nth beat from frame. Works with both positive and
+    /// negative values of n. If frame refers to the location of a beat,
+    /// then frame is returned. If no beat can be found, returns -1.
+    FramePos findNthBeat(FramePos frame, int offset) const;
+
+    int numBeatsInRange(FramePos startFrame, FramePos endFrame);
+
+    /// Find the frame N beats away from frame. The number of beats may be
+    /// negative and does not need to be an integer.
+    FramePos findNBeatsFromFrame(FramePos fromFrame, double beats) const;
+
+    /// Return an iterator to a container of Beats containing the Beats
+    /// between startFrameNum and endFrameNum. THe BeatIterator must be iterated
+    /// while a strong reference to the Beats object to ensure that the Beats
+    /// object is not deleted. Caller takes ownership of the returned BeatsIterator
+    std::unique_ptr<Beats::iterator> findBeats(FramePos startFrame,
+            FramePos stopFrame) const;
+
+    /// Return whether or not a Beat lies between startFrameNum and endFrameNum
+    bool hasBeatInRange(FramePos startFrame,
+            FramePos stopFrame) const;
+
+    /// Return the average BPM over the entire track if the BPM is
+    /// valid, otherwise returns -1
+    Bpm getBpm() const;
+
+    /// Return the average BPM over the range from startFrameNum to endFrameNum,
+    /// specified in frames if the BPM is valid, otherwise returns -1
+    double getBpmRange(FramePos startFrame,
+            FramePos stopFrame) const;
+
+    /// Return the average BPM over the range of n*2 beats centered around
+    /// curFrameNum.  (An n of 4 results in an averaging of 8 beats).  Invalid
+    /// BPM returns -1.
+    Bpm getBpmAroundPosition(FramePos curFrame, int n) const;
+
+    /// Sets the track signature at the nearest frame
+    void setSignature(TimeSignature signature, FramePos frame = FramePos());
+
+    /// Return the track signature at the given frame position
+    TimeSignature getSignature(FramePos frame = FramePos()) const;
+
+    /// Sets the nearest beat as a downbeat
+    void setDownBeat(FramePos frame = FramePos());
+
+    /// Add a beat at location frame.
+    void addBeat(FramePos frame);
```

Comment:
This function is only being used in a test and nowhere in the main program. It was proposed in 2512 as well to remove this. Are there any possible scenarios where someone would like to **add a beat**?
![Screenshot from 2020-06-18 22-22-41](https://user-images.githubusercontent.com/30816844/85635808-7a3a0e80-b69c-11ea-8bc1-a58f9f8fd9df.png)